src/backend/distributed/operations/citus_tools.c

```diff
@@ -60,6 +63,37 @@ static Tuplestorestate * CreateTupleStore(TupleDesc tupleDescriptor,
 										  StringInfo *resultArray, int commandCount);
 
 
+/*
+ * citus_check_connection_to_node sends a simple query from a worker node to another
+ * node, and returns success status.
+ */
+Datum
+citus_check_connection_to_node(PG_FUNCTION_ARGS)
+{
+	char *nodeName = PG_GETARG_TEXT_TO_CSTRING(0);
+	int32 nodePort = PG_GETARG_INT32(1);
+
+	bool success = CheckConnectionToNode(nodeName, nodePort);
+	PG_RETURN_BOOL(success);
+}
+
+
+/*
+ * CheckConnectionToNode sends a simple query from a node to another and returns success status
```

Comment:
Minor: As it may connect to itself to check own health, we can remove `another` word.

Reply:
Done.